libraries/chain/include/eosio/chain/abi_serializer.hpp

```diff
@@ -818,6 +842,84 @@ namespace impl {
+         }
+         if (vo.contains("ref_block_num")) {
+            from_variant(vo["ref_block_num"], trx.ref_block_num);
+         }
+         if (vo.contains("ref_block_prefix")) {
+            from_variant(vo["ref_block_prefix"], trx.ref_block_prefix);
+         }
+         if (vo.contains("max_net_usage_words")) {
+            from_variant(vo["max_net_usage_words"], trx.max_net_usage_words);
+         }
+         if (vo.contains("max_cpu_usage_ms")) {
+            from_variant(vo["max_cpu_usage_ms"], trx.max_cpu_usage_ms);
+         }
+         if (vo.contains("delay_sec")) {
+            from_variant(vo["delay_sec"], trx.delay_sec);
+         }
+         if (vo.contains("context_free_actions")) {
+            extract(vo["context_free_actions"], trx.context_free_actions, resolver, ctx);
+         }
+         if (vo.contains("actions")) {
+            extract(vo["actions"], trx.actions, resolver, ctx);
+         }
+
+         // can have "deferred_transaction_generation" (if there is a deferred transaction and the extension was "extracted" to show data),
+         // or "transaction_extensions" (either as empty or containing the packed deferred transaction),
+         // or both (when there is a deferred transaction and extension was "extracted" to show data and a redundant "transaction_extensions" was provided),
+         // or neither (only if extension was "extracted" and there was no deferred transaction to extract)
+         if (vo.contains("deferred_transaction_generation")) {
+            std::cerr << "has deferred_transaction_generation\n";
+            deferred_transaction_generation_context deferred_transaction_generation;
+            from_variant(vo["deferred_transaction_generation"], deferred_transaction_generation);
+            emplace_extension(
+               trx.transaction_extensions,
+               deferred_transaction_generation_context::extension_id(),
+               fc::raw::pack( deferred_transaction_generation )
+            );
+            // if both are present, they need to match
+            if (vo.contains("transaction_extensions")) {
+               std::cerr << "also has transaction_extensions\n";
+               extensions_type trx_extensions;
+               from_variant(vo["transaction_extensions"], trx_extensions);
+               EOS_ASSERT(trx.transaction_extensions == trx_extensions, packed_transaction_type_exception,
+                        "Transaction contained deferred_transaction_generation and transaction_extensions that did not match");
+            }
+         }
+         else if (vo.contains("transaction_extensions")) {
+            std::cerr << "has transaction_extensions\n";
+            from_variant(vo["transaction_extensions"], trx.transaction_extensions);
+         }
+      }
+
+      template<typename Resolver>
+      static void extract( const fc::variant& v, transaction& trx, Resolver resolver, abi_traverse_context& ctx )
+      {
```

Comment:
I think adding the static assert like add so new attributes will trip so someone will at least look at it would be good.

Reply:
Sorry, I meant literally add this: `static_assert(fc::reflector<transaction>::total_member_count == 9);`